UI/DisplayLayoutScreen.cpp

```diff
@@ -192,37 +191,87 @@ void DisplayLayoutScreen::CreateViews() {
+	rotation_->SetEnabledPtr(&displayRotEnable_);
+	displayRotEnable_ = (g_Config.iRenderingMode != FB_NON_BUFFERED_MODE);
+	bool bRotated = false;
+	if (g_Config.iRenderingMode != FB_NON_BUFFERED_MODE && (g_Config.iInternalScreenRotation == ROTATION_LOCKED_VERTICAL || g_Config.iInternalScreenRotation == ROTATION_LOCKED_VERTICAL180)) {	bRotated = true; }
 	mode_ = new ChoiceStrip(ORIENT_VERTICAL, new AnchorLayoutParams(leftColumnWidth, WRAP_CONTENT, 10, NONE, NONE, 158 + 64 + 10));
-	if (g_Config.iSmallDisplayZoom == 0) {
-		mode_->AddChoice(gr->T("Active (Auto)"));
-		float autoBound = bounds.w / 480.0f * 8.0f;
-		g_Config.fSmallDisplayCustomZoom = autoBound;
-		g_Config.fSmallDisplayOffsetX = 0.5f;
-		g_Config.fSmallDisplayOffsetY = 0.5f;
-	} else {
-		Choice *center = new Choice(di->T("Center"), "", false, new AnchorLayoutParams(leftColumnWidth, WRAP_CONTENT, 10, NONE, NONE, 84));
-		center->OnClick.Handle(this, &DisplayLayoutScreen::OnCenter);
-		root_->Add(center);
-		mode_->AddChoice(di->T("Move"));
-		mode_->AddChoice(di->T("Resize"));
-		mode_->SetSelection(0);
+	displayRepresentationScale_ = g_Config.fSmallDisplayZoomLevel * 8.0f; // Visual representation image is just icon size and have to be scaled 8 times to match PSP native resolution which is used as 1.0 for zoom
+	if (g_Config.iSmallDisplayZoomType > 1) { // Scaling
+		if (g_Config.iSmallDisplayZoomType == 2) { // Auto Scaling
+			mode_->AddChoice(gr->T("Auto Scaling"));
+			mode_->ReplaceLayoutParams(new AnchorLayoutParams(0, 0, local_dp_xres / 2.0f - 70.0f, NONE, NONE, local_dp_yres / 2.0f + 32.0f));
+			float autoBound = local_dp_yres / 270.0f;
+			// Case of screen rotated ~ only works with buffered rendering
+			if (bRotated) {
+				autoBound = local_dp_yres / 480.0f;
+			}
+			else { // Without rotation in common cases like 1080p we cut off 2 pixels of height, this reflects other cases
+				float resCommonWidescreen = autoBound - floor(autoBound);
+				if (resCommonWidescreen != 0.0f) {
+					float ratio = local_dp_xres / local_dp_yres;
+					if (ratio < orgRatio) {
+						autoBound = local_dp_xres / 480.0f;
+					}
+					else {
+						autoBound = local_dp_yres / 272.0f;
+					}
+				}
+			}
+			g_Config.fSmallDisplayZoomLevel = autoBound;
+			displayRepresentationScale_ = g_Config.fSmallDisplayZoomLevel * 8.0f;
+			g_Config.fSmallDisplayOffsetX = 0.5f;
+			g_Config.fSmallDisplayOffsetY = 0.5f;
+		} else { // Manual Scaling
+			Choice *center = new Choice(di->T("Center"), "", false, new AnchorLayoutParams(leftColumnWidth, WRAP_CONTENT, 10, NONE, NONE, 74));
+			center->OnClick.Handle(this, &DisplayLayoutScreen::OnCenter);
+			root_->Add(center);
+			PopupSliderChoiceFloat *zoomlvl_ = new PopupSliderChoiceFloat(&g_Config.fSmallDisplayZoomLevel, 1.0f, 10.0f, di->T("Zoom"), 1.0f, screenManager(), "* PSP res", new AnchorLayoutParams(leftColumnWidth, WRAP_CONTENT, 10, NONE, NONE, 134));
```

Comment:
Does `"* PSP res"` need a `->T()`?  Not sure if it's translated in `PopupSliderChoiceFloat` or not...

-[Unknown]